### DISCUSSION TITLE
Request for information on cross compiling qdrant for arm64

### DISCUSSION THREAD

**Discussion by cravishe** [qdrant/qdrant#3434] [General]
Hello,
I'm building qdrant for arm64 and facing an error while building [api ](https://github.com/qdrant/qdrant/tree/master/lib/api) library. The issue is due to [ring ](https://github.com/briansmith/ring) crate. However, I have made changes after cloning ring and I'm able to cross-compile it. Can I know how and where ring is used in api (directly or indirectly) so that I can use locally built ring to compile api module. Any information on this will be helpful.

Thank you

**Comment by generall** [comment]
Qdrant provides a docker image built for arm64 in the default dockerhub registry. Commands used in default Dockerfile should be enough to build arm64 binary.

**Comment by cravishe** [comment]
> Qdrant provides a docker image built for arm64 in the default dockerhub registry. Commands used in default Dockerfile should be enough to build arm64 binary.

Thank you for the reply. I'm trying to build from source for arm64 and not planning to use the docker image..

**Reply by generall** [reply]
Sure, still all the configurations needed to build from source could be found in Dockerfile
